skyportal/tests/frontend/test_followup_requests.py

```diff
@@ -0,0 +1,201 @@
+import uuid
+import pytest
+import requests
+from selenium.common.exceptions import TimeoutException
+from selenium.webdriver.common.action_chains import ActionChains
+from baselayer.app.env import load_env
+from skyportal.tests import api
+
+env, cfg = load_env()
+endpoint = cfg['app.sedm_endpoint']
+sedm_isonline = requests.get(endpoint, timeout=5).status_code in [200, 400]
+
+
+def add_telescope_and_instrument(instrument_name, token):
+    status, data = api("GET", f"instrument?name={instrument_name}", token=token)
+    if len(data["data"]) == 1:
+        return data["data"][0]
+
+    telescope_name = str(uuid.uuid4())
+    status, data = api(
+        "POST",
+        "telescope",
+        data={
+            "name": telescope_name,
+            "nickname": telescope_name,
+            "lat": 0.0,
+            "lon": 0.0,
+            "elevation": 0.0,
+            "diameter": 10.0,
+            "robotic": True,
+        },
+        token=token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+    telescope_id = data["data"]["id"]
+
+    status, data = api(
+        "POST",
+        "instrument",
+        data={
+            "name": instrument_name,
+            "type": "imager",
+            "band": "Optical",
+            "telescope_id": telescope_id,
+            "filters": ["ztfg"],
+            "api_classname": "SEDMAPI",
+        },
+        token=token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+    return data["data"]
+
+
+def add_allocation(instrument_id, group_id, token):
+    status, data = api(
+        "POST",
+        "allocation",
+        data={
+            "group_id": group_id,
+            "instrument_id": instrument_id,
+            "hours_allocated": 100,
+            "pi": "Ed Hubble",
+        },
+        token=token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+    return data["data"]
+
+
+def _followup_request_test_body(
+    driver, super_admin_user, public_source, super_admin_token, public_group
+):
+    """Adds a new followup request and makes sure it renders properly."""
+
+    idata = add_telescope_and_instrument("SEDM", super_admin_token)
+    add_allocation(idata['id'], public_group.id, super_admin_token)
+
+    driver.get(f"/become_user/{super_admin_user.id}")
+    driver.get(f"/source/{public_source.id}")
+
+    try:
+        # wait for the plots to load
+        driver.wait_for_xpath(
+            '//div[@class="bk-root"]//span[text()="Flux"]'
+        )  # waits for photometry plot
+        driver.wait_for_xpath(
+            '//div[@class="bk-root"]//label[text()="Mg"]'
+        )  # waits for spectroscopy plot
+    except TimeoutException:
```

Comment:
What's happening here?

Reply:
Waiting for the plots to load before trying to select things on the FollowupForm. Sometimes the plots dont load within the timeout window so I make a second attempt.

Reply:
"Mg" is one of the elements I look for in the spectroscopy plot to indicate that it has loaded

Reply:
I can see the point of long timeouts *or* of repeating the procedure several times, but they don't make sense to me in combination.

Reply:
I think I have isolated the issue here, so will pare this down.